File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <iostream>
using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    vector<int> cmp;
    cmp.push_back(nums[0]);
    
    for(int i = 1; i<nums.size(); ++i) {
        bool duplicated = false;
        for(int j=0; j<cmp.size(); ++j) 
            if(nums[i] == cmp[j]) duplicated = true;

        if(!duplicated) { cmp.push_back(nums[i]); }
    }
    
    for(int i=0; i<cmp.size();++i) {
        cout<<cmp[i]<<endl;
    }
    if(cmp.size()>(nums.size()/2)) return (nums.size()/2);
    else return cmp.size();
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        int size = 0;
        
        Set<Integer> hs = new HashSet<>();
    
        
        for(int i : nums) {
            hs.add(i);
            size++;
        }
        
        if(size / 2 < hs.size()) {
            answer = size / 2;
        } else {
            answer = hs.size();
        }
        return answer;
    }
}